PROJECTS/SDSS/PS1QLS.py

```diff
@@ -121,7 +121,44 @@ def make_roc(self,truth):
         plt.plot(fpr,tpr,'b--')
         plt.xlabel('FPR')
         plt.ylabel('TPR')
-        return fpr,tpr
+        return fpr, tpr
+    
+    def make_pc_plot(self, truth, abundance=1e-3):
+        """
+        Plots purity against completeness, assuming a natural abundance of lenses.
+        
+        Parameters
+        ----------
+        truth: ndarray
+            True labels for the test dataset objects.
+        abundance: float
+            Fraction of all objects in the expected parent dataset that are actually lenses.
+        // code starts here
+        
+        precision, recall, _ = sklearn.metrics.precision_recall_curve(truth, self.lQSO_probs, pos_label=1)
+        fpr, tpr, _ = metrics.roc_curve(truth,self.lQSO_probs,pos_label=1)
```

Comment:
Should this be `sklearn.metrics.roc_curve` to match the line above?
I guess both functions have the same defaults for the number of threshold points to plot, right? To defend against unexpected behavior you could `assert len(recall) == len(tpr) at line 140, just in case - or even better, pass the two metrics functions the same input kwarg for this number so that the arrays all come out the same length.